proxy/http/HttpSM.cc

```diff
@@ -1800,6 +1798,65 @@ HttpSM::handle_api_return()
   }
 }
 
+void
+HttpSM::set_server_txn(ProxyTransaction *txn)
+{
+  ink_release_assert(server_txn == nullptr);
```

Comment:
I would probably argue that this should be a debug assert, it kinda feels like we're too liberal sometimes with the release assert. Looking where this is called, it can't be nullptr (assuming that to_return->new_transaction() is not allowed to fail).

Reply:
Agree.  I use the release_asserts liberally while debugging in production.  Don't always get them cleaned up.  Looking forward to using your enable-all asserts option.